aya-log/src/lib.rs

```diff
@@ -112,12 +114,54 @@ impl EbpfLogger {
         bpf: &mut Ebpf,
         logger: T,
     ) -> Result<EbpfLogger, Error> {
-        let logger = Arc::new(logger);
-        let mut logs: AsyncPerfEventArray<_> = bpf
-            .take_map(MAP_NAME)
-            .ok_or(Error::MapNotFound)?
-            .try_into()?;
+        let map = bpf.take_map(MAP_NAME).ok_or(Error::MapNotFound)?;
+        Self::read_logs_async(map, logger)?;
+        Ok(EbpfLogger {})
+    }
+
+    /// Attaches to an existing `aya-log-ebpf` log created by a running ebpf
+    /// program identified by `program_id`. The log records will be written
+    /// to the default logger. See [log::logger].
+    /// It can be used to review the log for a pinned program after the user
```

Comment:
nit: please add an empty line before this one. Separate paragraphs must be split by empty lines.

Also it should say "It can be used to read logs generated by a pinned program."

Review is not the correct word here, and the application doesn't necessarily have to exit, you could read logs of an application that is still running but wants another process to process for whatever reason (centralized logs for example).

Reply:
I've used _review_ in the sense that someone or some process would want to _examine_ the logs later. I changed it to _read_ to not confuse about the meaning.
You're right it's not necessarily that the first app to exit.